libraries/chain/include/eosio/chain/webassembly/binaryen.hpp

```diff
@@ -157,6 +157,7 @@ class binaryen_runtime : public eosio::chain::wasm_runtime_interface {
 template<typename T>
 inline array_ptr<T> array_ptr_impl (interpreter_interface* interface, uint32_t ptr, uint32_t length)
 {
+   FC_ASSERT( length < INT_MAX/(uint32_t)sizeof(T), "length will overflow" );
```

Comment:
Is there a reason to have `INT_MAX` vs unsigned int max?  Also can this be `<=` ?  the division will truncate down so if length == a/b then length * b should be <= a right?

Reply:
in theory we should be nowhere near INT_MAX and bart's suggestion would only serve to increase the range to the maximum "safe" limit.  

Since this is suppose to be a sanity check, I think it is already conservative. No WASM module should address more than 32MB of RAM.